.github/workflows/artifacts_CI.yml

```diff
@@ -0,0 +1,112 @@
+        with:
+          version: ${{ matrix.version }}
+          arch: ${{ matrix.arch }}
+      - uses: julia-actions/cache@v1
+      # TODO: We shouldn't require separate caches per Julia version but allow cache restores to work
+      # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      # https://github.com/beacon-biosignals/Ray.jl/issues/63
+      - name: Restore build cache
+        uses: actions/cache/restore@v3
+        id: build-cache
+        with:
+          key: build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
+          path: ~/.cache
+          restore-keys: |
+            build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
+            build-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
+
+      # Based upon:
+      # https://docs.ray.io/en/releases-2.5.1/ray-contribute/development.html#building-ray-on-linux-macos-full
+      - name: Build Ray CLI
+        run: |
+          pip install --upgrade pip wheel
+
+          # The Ray BUILD.bazel includes a bunch of `copy_to_workspace` rules which copy build output
+          # into the Ray worktree. When we only restore the Bazel cache then re-building causes these
+          # rules to be skipped resulting in `error: [Errno 2] No such file or directory`. By manually
+          # saving/restoring these files we can work around this.
+          RAY_GEN_CACHE_DIR=~/.cache/ray-generated
+          if [ -d "$RAY_GEN_CACHE_DIR" ]; then
+              dest=$(pwd)
+              pushd $RAY_GEN_CACHE_DIR
+              cp -rp --parents \
+                  python/ray/_raylet.so \
+                  python/ray/core/generated \
+                  python/ray/serve/generated \
+                  python/ray/core/src/ray/raylet/raylet \
+                  python/ray/core/src/ray/gcs \
+                  $dest
+              popd
+          fi
+
+          pushd python
+          pip install . --verbose  # Fresh build takes ~50 minutes on basic GH runner
+          popd
+
+          # By copying the entire Ray worktree we can easily restore missing files without having to
+          # delete the cache and build from scratch.
+          mkdir -p "$RAY_GEN_CACHE_DIR"
+          cp -rfp . "$RAY_GEN_CACHE_DIR"
+        working-directory: ${{ env.ray_dir }}
+      - name: "Monorepo setup"
+        shell: julia --color=yes --project {0}
+        run: |
+          overrides_toml = joinpath(first(Base.DEPOT_PATH), "artifacts", "Overrides.toml")
+          isfile(overrides_toml) && rm(overrides_toml)
+          using Pkg
+          Pkg.develop(PackageSpec(; path="${{ env.ray_julia_jll_dir }}"))
+      - uses: julia-actions/julia-buildpkg@v1
+      - uses: julia-actions/julia-runtest@v1
```

Comment:
I wanted to run tests for ray_julia_jll as well. I Was gonna make a separate workflow (file) for that but it's so boiler-platey I was tempted to just tack it on the end here?

Reply:
You can just do that as part of the same job using `julia-runtest` with another project

Reply:
thought so! thanks